### ISSUE TITLE
Lint reader syntax errors in #_ ignored forms

### ISSUE THREAD

**Issue by yuhan0** [borkdude/clj-kondo#929]
**Is your feature request related to a problem? Please describe.**
The Clojure reader requires map and set literals to be syntatically correct even when they are ignored with the #_ reader macro. None of the following compile because of duplicate keys or odd number of forms.

```clj
#_ #{1 2 3 1} 
#_ {:a 1 :a 2}
#_ {:a}
```
However clj-kondo doesn't pick up on these syntax errors, which end up being very tricky to pinpoint in a large nested form,  since the error stacktraces does not include location info, and editors tend to gray-out ignored forms making them hard to notice.

**Describe the solution you'd like**
clj-kondo should detect reader errors in #_ ignored forms, using the same error messages it does for regular forms.

**Comment by yuhan0** [comment]
More examples of read errors:

```clj
#_ ::unresolved/alias
#_ #::unresolved{:map :alias}
#_ #(#())
```

For some reason clj-kondo does catch invalid numeric literals:
```clj
#_ 0Z
#_ 1.0/2
```

**Comment by borkdude** [comment]
Still interesting
